src/main/java/org/myrobotlab/service/config/ServiceConfig.java

```diff
@@ -1,26 +1,100 @@
 package org.myrobotlab.service.config;
 
+import java.lang.reflect.Constructor;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.myrobotlab.framework.Peer;
+import org.myrobotlab.framework.Plan;
+import org.myrobotlab.logging.LoggerFactory;
+import org.myrobotlab.service.Runtime;
+import org.slf4j.Logger;
+
 /**
  * Base service configuration class. All services must have a type. The name of
  * the service config file implies the name of the service.
+ * 
+ * FIXME - make a 2 param constructor that makes the "default" callback name
  *
  */
 public class ServiceConfig {
 
+  public final static Logger log = LoggerFactory.getLogger(ServiceConfig.class);
+  
+  
+  public static class Listener {
```

Comment:
This looks essentially identical to `MRLListener`; what's the reasoning behind duplicating it? Also missing `equals()` but has `hashCode()`, if one is there it's always a good idea to have the other.

Reply:
Its for specifically configuration .. MRLListener will not look as clean when serialized, and I don't want to modify it.